rt/rs/microprofile-client/pom.xml

```diff
@@ -134,15 +134,20 @@
         <dependency>
             <groupId>org.eclipse.microprofile.config</groupId>
             <artifactId>microprofile-config-api</artifactId>
+            <scope>provided</scope>
```

Comment:
@brevilo @rmannibucau Guys, I think `<scope>provided</scope>` is redundant here,  `<optional>true</optional>` is suffice by all means. Why:
 - `<scope>provided</scope>` means that dependency is **required** and has to be provided at runtime [1]
 - `<optional>true</optional>` means dependency is **not required** but for some features only [2]
 
[1] https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html
[2] https://maven.apache.org/guides/introduction/introduction-to-optional-and-excludes-dependencies.html

Reply:
True until you use either a plugin which uses only one or a scraper/lib/tool which reads the pom and only uses one.
Strictly speaking (maven), provided means "will be there at compile time but not transitively", nothing more (rest is wording) and it was originally intended for specs (like here funnily) but optional was added (for OSGi I guess) to do the same so as of today, if the pom is consumed (and it is for CXF) I guess the best is to have both (minimal being provided which is the most supported one I guess?).

Reply:
Fair enough, let's go with both than :)